repository/src/main/resources/db/changelog/db.changelog-rdm-2917.xml

```diff
@@ -0,0 +1,504 @@
+
+        <!-- START - role table -->
+        <!-- create the new role table -->
+        <createTable tableName="role">
+            <column name="id" type="serial">
+                <constraints nullable="false"/>
+            </column>
+            <column name="reference"  type="varchar(40)" >
+                <constraints nullable="false" />
+            </column>
+            <column name="name" type="varchar(40)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="description"  type="varchar(128)" >
+                <constraints nullable="true" />
+            </column>
+            <column name="case_type_id" type="integer">
+                <constraints nullable="true"/>
+            </column>
+            <column name="created_at" type="datetime" defaultValue="now()">
+                <constraints nullable="false"/>
+            </column>
+        </createTable>
+
+        <addPrimaryKey columnNames="id"
+                       constraintName="pk_role"
+                       tableName="role"/>
+
+        <addForeignKeyConstraint baseColumnNames="case_type_id"
+                                 baseTableName="role"
+                                 constraintName="fk_role_case_type_id"
+                                 referencedColumnNames="id"
+                                 referencedTableName="case_type"/>
+
+        <addUniqueConstraint columnNames="case_type_id, reference"
+                             constraintName="unique_role_case_type_id_role_reference"
+                             tableName="role"/>
+
+        <!-- Add foreign key for case_type_id column linking to case_type table -->
+        <addForeignKeyConstraint baseColumnNames="case_type_id"
+                                 baseTableName="role"
+                                 constraintName="fk_role_case_type_id_case_type_id"
+                                 referencedColumnNames="id"
+                                 referencedTableName="case_type"/>
+
+        <!-- Change precision of reference, name and description columns in role table -->
+        <modifyDataType columnName="reference" newDataType="varchar(255)" tableName="role"/>
+        <modifyDataType columnName="name" newDataType="varchar(255)" tableName="role"/>
+        <modifyDataType columnName="description" newDataType="varchar(255)" tableName="role"/>
+
+        <!-- Add new security_classification, user_role_id and dtype fields to the role table -->
+        <addColumn tableName="role">
+            <column name="user_role_id" type="integer" >
+                <constraints nullable="true"/>
+            </column>
+            <column name="security_classification" type="security_classification" defaultValue="PUBLIC">
+                <constraints nullable="true"/>
+            </column>
+            <column name="dtype" type="varchar(10)">
+                <constraints nullable="false"/>
+            </column>
+        </addColumn>
```

Comment:
For clarity, those modifications should be merged inside the `createTable` statement for table `role`